README.md

```diff
@@ -1,11 +1,119 @@
 # tsc-ls
 
-TypeScript compiler using language service plugins.
+TypeScript compiler overlay using language service plugins.
 
 [![npm](https://img.shields.io/npm/v/tsc-ls)](https://www.npmjs.com/package/tsc-ls)
 [![license](https://img.shields.io/npm/l/tsc-ls)](https://github.com/chnapy/tsc-ls/blob/master/LICENSE)
 
-TODO
+One of TypeScript compiler limitation is its inability to use language services plugins during compilation.
+[By design](https://github.com/microsoft/TypeScript/wiki/Using-the-Language-Service-API), language services are meant to be used by editors only.
 
-- compiler running every tsconfig plugins `init(params1).create(params2)`
-- allows params customization
+This limitation makes not possible type-checking using some plugins in CLI environments, like with CIs.
+
+A good example would be with [`typescript-plugin-css-modules`](https://github.com/mrmckeb/typescript-plugin-css-modules) plugin, which allows to add CSS classes typing:
+
+```json
+// tsconfig.json
+{
+  "compilerOptions": {
+    "composite": true,
+    "plugins": [
+      {
+        "name": "typescript-plugin-css-modules"
+      }
+    ]
+  },
+  "include": ["."]
+}
+```
+
+```css
+/* toto.module.css */
+
+.button {
+  color: red;
+}
+```
+
+```ts
+import styles from './toto.module.css';
+
+styles.button; // OK
+styles.toto; // Editor shows type error
+```
+
+With this example, editors like VSCode shows correctly the error. But if you use `tsc` to compile your file, it will emit no error.
+
+This is why `tsc-ls` was made, to make compilation using language service plugins. So it shows errors in the console like it should.
+
+# Limitations
+
+This tool is young, tested but only for some cases, nothing is guaranteed.
+
+This inconsistency is due to TypeScript environment, which does not allow to easily work with its compiler. Like said upper, language services were not meant to be used like that, only by editors.
+Like with some popular plugins, multiple globals may be mutated during process. `tsc-ls` cannot know what plugins will do, so it's not possible to offer a totally generic solution working with every plugins.
+
+For now it was tested with these plugins:
+
+- [`typescript-plugin-css-modules`](https://github.com/mrmckeb/typescript-plugin-css-modules)
+- [`ts-gql-plugin`](https://github.com/Chnapy/ts-gql-plugin)
+
+Using others plugins may not work. In this case consider open an issue.
+
+Also only some of possible TypeScript projects configurations are handled, it may not work with your projects - again here, consider open an issue.
```

Comment:
```suggestion
Also, only some TypeScript project configurations are handled, it may not work with your projects - again here, consider opening an issue.
```